brave/brave-core
# Add Rewards metrics for page & panel views

Resolves https://github.com/brave/brave-browser/issues/35146



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

For `Brave.Rewards.DesktopPanelCount` (only test on desktop): 

0. Start with fresh profile, ensure metric is not present in local state page.
1. Open the Rewards panel multiple times, ensure metric still does not exist.
2. Enable rewards. Open the Rewards panel once. Ensure metric has value `0`.
3. Open the Rewards panel multiple times, ensure metric value corresponds to the amount of panel opens.

In addition, ensure this metric does not show up on Android.

For `Brave.Rewards.PageViewCount` (test on desktop & Android):

0. Start with fresh profile, ensure metric is not present in local state page.
1. Access `brave://rewards` in a new tab, ensure metric still does not exist.
2. Enable rewards. Open `brave://rewards` in a new tab once. Ensure metric has value `0`.
3. Open the Rewards page multiple times in new tabs, ensure metric value corresponds to the amount of page opens.
